### PR TITLE
Add config props for adaptive remote task req size

### PR DESCRIPTION
**Pull request by blvckcoffee**
## Description

- Adds documentation for adaptive remote task request size to Query management properties

## Additional context and related issues

- first pass at documentation for: https://github.com/trinodb/trino/pull/15721

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 1]
```diff
@@ -171,6 +171,24 @@ The minimal age of a query in the history before it is expired. An expired
...
+Enables dynamic split of ``taskUpdateRequest`` for schemas featuring large
+strings in table properties.
+
+``query.remote-task.guaranteed-splits-per-task``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Default value:** ``3``
+
+The number of guaranteed splits that will be sent in ``taskUpdateRequest`` for
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-query-management.rst:189]:
```suggestion
The number of guaranteed splits that is sent in ``taskUpdateRequest`` for
```

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 2]
```diff
@@ -181,6 +199,25 @@ Timeout value for remote tasks that fail to communicate with the
 coordinator. If the coordinator is unable to receive updates from a remote task
 before this value is reached, the coordinator treats the task as failed.
 
+``query.remote-task.max-request-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Default value:** ``8``
+
+The maximum size of ``taskUpdateRequest`` in megabytes.
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-query-management.rst:208]: I dont think this is in megabytes ..

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 3]
```diff
@@ -181,6 +199,25 @@ Timeout value for remote tasks that fail to communicate with the
 coordinator. If the coordinator is unable to receive updates from a remote task
 before this value is reached, the coordinator treats the task as failed.
 
+``query.remote-task.max-request-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Default value:** ``8``
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-query-management.rst:206]: thats not a valid data size value

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 4]
```diff
@@ -181,6 +199,25 @@ Timeout value for remote tasks that fail to communicate with the
...
+
+The maximum size of ``taskUpdateRequest`` in megabytes.
+
+``query.remote-task.request-size-headroom``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Default value:** ``2``
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-query-management.rst:214]: not a valid value

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 5]
```diff
@@ -181,6 +199,25 @@ Timeout value for remote tasks that fail to communicate with the
...
+
+The headroom available, in megabytes, for ``taskUpdateRequest`` where the
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-query-management.rst:216]: not in megabytes

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 6]
```diff
@@ -181,6 +199,25 @@ Timeout value for remote tasks that fail to communicate with the
...
+expected mean value of ``taskUpdateRequest`` will be ``request-size-limit`` -
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/properties-query-management.rst:217]:
```suggestion
expected mean value of ``taskUpdateRequest`` is ``request-size-limit`` -
```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 7]
```diff
@@ -87,7 +87,7 @@ memory availability. Supports the following values:
 
 The amount of time a query is allowed to recover between running out of memory
 and being killed, if ``query.low-memory-killer.policy`` or
-``task.low-memory-killer.policy`` is set to value differnt than ``none``.
+``task.low-memory-killer.policy`` is set to value different than ``none``.
```
Comment (colebow) [inline docs/src/main/sphinx/admin/properties-query-management.rst:90]: Because this is an unrelated change, this should be pulled out into a separate commit.

File: docs/src/main/sphinx/admin/properties-query-management.rst (hunk 1 above)
Comment (colebow) [reply docs/src/main/sphinx/admin/properties-query-management.rst:189]: are...

File: docs/src/main/sphinx/admin/properties-query-management.rst (hunk 3 above)
Comment (colebow) [reply docs/src/main/sphinx/admin/properties-query-management.rst:206]: Yeah, this should be 8MB, and you don't need to specify that it's in megabytes. A data size is `2GB` or `16MB` or `10KB`, the user specifies their own units.

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 8]
```diff
@@ -171,6 +171,24 @@ The minimal age of a query in the history before it is expired. An expired
 query is removed from the query history buffer and no longer available in
 the :doc:`/admin/web-interface`.
 
+``query.remote-task.enable-adaptive-request-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``true``
+
+Enables dynamic split of ``taskUpdateRequest`` for schemas featuring large
```
Comment (colebow) [inline docs/src/main/sphinx/admin/properties-query-management.rst:180]: taskUpdateRequest isn't a user-visible construct. I'd only mention that it's dynamically splitting tasks.

You can also say "large schemas." A schema is the shape of the table, and there's no need to say "schema string."

So my proposal:
"Enables dynamically splitting up server requests sent by tasks, which can prevent out-of-memory errors for large schemas."

**Comment by blvckcoffee** [comment]
Thanks for the initial feedback. I have made changes according to your recommendations. @mosabua @colebow

File: docs/src/main/sphinx/admin/properties-query-management.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/admin/properties-query-management.rst]: Hahah ... one number is.. many splits are .. ;-)

File: docs/src/main/sphinx/admin/properties-query-management.rst [hunk 9]
```diff
@@ -181,6 +199,25 @@ Timeout value for remote tasks that fail to communicate with the
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Default value:** ``2MB``
+
+The headroom available for ``taskUpdateRequest`` where the
+expected mean value of ``taskUpdateRequest`` is ``request-size-limit`` -
+``request-size-limit-headroom``.
+
+
```
Comment (jhlodin) [inline docs/src/main/sphinx/admin/properties-query-management.rst:220]: Remove extra line

**Comment by mosabua** [comment]
We discussed this with @electrum and also looked at the code some more. @blvckcoffee is updating accordingly and then we will work with @arosa on user facing explanation.

**Review by mosabua — APPROVED** [review]
Looks good to me.

**Comment by mosabua** [comment]
@arhimondr can you merge this please?

**Review by arhimondr — APPROVED** [review]
